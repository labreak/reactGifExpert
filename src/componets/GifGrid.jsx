import { GifItem } from "./GifItem";
import { useFetchGifs } from '../hooks/useFetchGifs';
import  PropTypes  from 'prop-types';


export const GifGrid = ({categoria}) => {

const {images,isLoading} =useFetchGifs( categoria);


  return (

<>
<h2>{categoria}</h2>
{
  isLoading && (<p>Cargando ...</p> )
}


<div className='card-grid'>
{images.map(image =>
            (
               <GifItem key={image.id} {...image}
               />
               
                    
               
            )
            )
            }

</div>


</>
    
  )
}

GifGrid.propTypes = {
  categoria : PropTypes.string.isRequired,
}
