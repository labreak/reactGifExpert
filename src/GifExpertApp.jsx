import { useState } from 'react';
import { AddCategory, GifGrid } from './componets';


export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['Dragon Ball']);

    const onAddCategory = (newCategory) => {

        if (categories.includes(newCategory)) return;

        setCategories([newCategory, ...categories]);


    }

    return (
        <>


            <h1>GifExpertApp</h1>

            <AddCategory

                onNewCategory={(value) => onAddCategory(value)}
            />



            {categories.map(categoria =>
            (
                <GifGrid
                    key={categoria}
                    categoria={categoria}
                />
            )
            )
            }




        </>
    )




}