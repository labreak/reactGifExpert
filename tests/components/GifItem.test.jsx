import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/componets";

describe('Pruebas en <GifItem/>', () => {

    const title = "Hola Soy Nicolas";
    const url="https://nico-test.com/lii.jpg";
  
  
    test('Debe hacer match con el snapshot', () => {
  
      const { container } = render(<GifItem title={title} url={url}/>)
     
      expect(container).toMatchSnapshot();
  
    });
  
    


    test('Debe mostrar imagen con url y alt indicado', () => {
  
      render(<GifItem title={title} url={url} />)
      expect(screen.getByRole('img').src).toEqual(url);
      expect(screen.getByRole('img').alt).toEqual(title);

      const {src,alt} =screen.getByRole('img');
      expect(src).toBe(url)
      expect(alt).toBe(title)
     
      
  
    });

    test('debe de mostrar el titulo en el componente', () => {

      render(<GifItem title={title} url={url} />)

      expect(screen.getByText(title)).toBeTruthy();

      })
 
  
  });