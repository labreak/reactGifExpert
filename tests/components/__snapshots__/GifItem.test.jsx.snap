// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem/> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Hola Soy Nicolas"
      src="https://nico-test.com/lii.jpg"
    />
    <p>
      Hola Soy Nicolas
    </p>
  </div>
</div>
`;
